### PR TITLE
[Enhance] try to release chunks of operator when close

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

When `Operator::close`, it could release the buffered chunk to avoid occupy too much memory.

### CODE REVIEW

**Review by stdpain — DISMISSED** [review]

File: be/src/exec/pipeline/crossjoin/cross_join_context.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ class CrossJoinContext final : public ContextWithDependency {
               _rf_hub(params.rf_hub),
               _rf_descs(std::move(params.rf_descs)) {}
 
-    void close(RuntimeState* state) override {}
+    void close(RuntimeState* state) override { _build_chunks.clear(); }
```
Comment (imay) [inline be/src/exec/pipeline/crossjoin/cross_join_context.h:38]: Better to implement this function in source files, in order to reduce the header dependence of this file.

File: be/src/exec/pipeline/sort/sort_context.h [hunk 2]
```diff
@@ -36,7 +36,10 @@ class SortContext final : public ContextWithDependency {
         _chunks_sorter_partions.reserve(num_right_sinkers);
     }
 
-    void close(RuntimeState* state) override {}
+    void close(RuntimeState* state) override {
```
Comment (imay) [inline be/src/exec/pipeline/sort/sort_context.h:39]: ditto

File: be/src/exec/pipeline/crossjoin/cross_join_context.h (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exec/pipeline/crossjoin/cross_join_context.h:38]: done

File: be/src/exec/pipeline/sort/sort_context.h (hunk 2 above)
Comment (mofeiatwork) [reply be/src/exec/pipeline/sort/sort_context.h:39]: done

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2 branch-2.3

**Review by satanson — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Review by imay — DISMISSED** [review]

**Review by ZiheLiu — DISMISSED** [review]

**Review by trueeyu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**